### PR TITLE
7503 request asking price

### PR DESCRIPTION
**Pull request by RemcoSimonides**
#7503

### CODE REVIEW

File: libs/movie/src/lib/movie/components/upcoming-screenings/upcoming-screenings.component.html [hunk 1]
```diff
@@ -38,12 +44,17 @@ <h5>{{ sessions[index] }} Session</h5>
...
+
+      <section fxLayout="column" fxLayoutAlign="center center" fxLayoutGap="12px">
+        <ng-container *ngIf="buttonState$ | async else noInvitation">
+          <button mat-flat-button disabled>Already applied for this Session</button>
+        </ng-container>
+        <ng-template #noInvitation>
+          <button mat-flat-button color="accent" (click)="askForInvitation(futureScreenings)">Register for this session</button>
+        </ng-template>
+        <button mat-button (click)="requestAskingPrice()">Request Asking Price</button>
+      </section>
```
Comment (RemcoSimonides) [inline libs/movie/src/lib/movie/components/upcoming-screenings/upcoming-screenings.component.html:56]: The three `Request asking price` buttons on this page, I can do better. Could be only one button but layout was ugly so I'd have to rework the css a bit.

File: apps/backend-functions/src/movie.ts [hunk 2]
```diff
@@ -194,6 +197,45 @@ export async function onMovieUpdate(
   }
 }
 
+export async function createAskingPriceRequest(data: { uid: string, movieId: string, territories: string, message: string }, context: CallableContext) {
+  const { uid, movieId, territories, message } = data;
+
+  if (!context?.auth) { throw new Error('Permission denied: missing auth context.'); }
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/movie.ts:203]: you don't need the `{ }`

File: libs/movie/src/lib/movie/components/request-asking-price/request-asking-price.component.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    </static-group>
+    <mat-hint>Choose territories you are interested in.</mat-hint>
+    <mat-error *ngIf="form.get('territories').hasError('required')">Select at least one territory</mat-error>
+  </mat-form-field>
+
+  <mat-form-field appearance="outline">
+    <mat-label>Your message</mat-label>
+    <textarea
+      matInput
+      placeholder=""
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/request-asking-price/request-asking-price.component.html:26]: remove placeholder if you don't put anything

File: libs/movie/src/lib/movie/components/upcoming-screenings/upcoming-screenings.component.html [hunk 4]
```diff
@@ -6,7 +6,10 @@
       <ng-container *ngFor="let org of orgs$ | async">
         <h3>There is no screening scheduled at the moment.</h3>
         <p class="mat-body-1">You can contact <a [routerLink]="['/c/o/marketplace/organization/', org.id]">{{ org.denomination?.public || 'the organization' }}</a> for any questions.</p>
-        <event-request-screening [movieId]="movieId"></event-request-screening>
+        <section fxLayout="column" fxLayoutAlign="center center" fxLayoutGap="12px">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/upcoming-screenings/upcoming-screenings.component.html:9]: we usually not use 12px as a value for margin. Either 8px or 16px, except if @stephanie-cascade8 says so.

File: libs/movie/src/lib/movie/components/upcoming-screenings/upcoming-screenings.component.html (hunk 1 above)
Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/components/upcoming-screenings/upcoming-screenings.component.html:56]: looks fine to me

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**